File: leetcode/leet179.py
# py2
# largest-number

class Solution(object):
  def largestNumber(self, nums):
    """
    :type nums: List[int]
    :rtype: str
    """
    res = []
    for i in nums:
      res.append(str(i))
    res = self.sort(res)
    res = ''.join(res)
    res = str(int(res))
    print res
    return res
  def compareMax(self, val1, val2):
    """
    :type val1: str
    :type val2: str
    :rtype: bool
    """
    n1 = int(val1+val2)
    n2 = int(val2+val1)
    resBool = n1 < n2
    return resBool
  def sort(self, list):
    """
    :type list: List[str(int)]
    :rtype: List[str(int)]
    """
    newList = list
    listLen = len(newList)
    if listLen > 1:
      for i in range(listLen-1):
        for j in range(listLen-i-1):
          if self.compareMax(newList[j], newList[j+1]):
            newList[j], newList[j+1] = newList[j+1], newList[j]
          j += 1
        i += 1
    return newList

if __name__ == "__main__":
  testData = [
    [10,2], # 210
    [3,30,34,5,9], # 9534330
    [0,0],
  ]
  test = Solution()
  for i in testData:
    print i
    test.largestNumber(i)
  

File: leetcode/level2/leet179.js


/**
 * @param {number[]} nums
 * @return {string}
 */
// var largestNumber = function(nums) {
//     nums.sort((a, b) => a+''+b > b+''+a ? -1 : 1)
//     return nums.join('')
// };
var largestNumber = function(nums) {
    nums.sort((a, b) => a+''+b > b+''+a ? -1 : 1)
    return nums.every(v => v===0) ? '0' : nums.join('') 
};

var testData = [
    [10,2],
    [3,30,34,5,9],
    [0,0],
    [1,2,3,4,5,6,7,8,9,0]
]

for (let v of testData) {
    console.log(largestNumber(v));
}